sched/sched/sched_waitid.c

```diff
@@ -143,6 +143,14 @@ int nx_waitid(int idtype, id_t id, FAR siginfo_t *info, int options)
   sigemptyset(&set);
   nxsig_addset(&set, SIGCHLD);
 
+  /* NOTE: sched_lock() is not enough for SMP
+   * because the child task is running on another CPU
+   */
+
+#ifdef CONFIG_SMP
+  irqstate_t flags = enter_critical_section();
```

Comment:
Isn't the call to `sched_(un)lock` not needed then?

Reply:
@Ouss4 
Thanks for your comment. I will modify the code and do the regression test again.

Reply:
@Ouss4 
I tried the following change in sched_waitid.c and sched_waitpid.c
The ostest passed for the all SMP configurations.

However, Wi-Fi streaming plus stress testing with spresense:wifi_smp (NCPUS=4, DEBUG_ASSERTIONS=y)
failed with DEBUGASSERTION after 3.5 hours. (FYI, without the following changes, the test continued over 9 hours)

I think replacing sched_lock() and sched_unlock() with a critical section should be a future task.
What do you think?

```
#ifdef CONFIG_SMP                                                                                                                                                                   
  /* NOTE: sched_lock() is not enough for SMP                                                                                                                                       
   * because the child task is running on another CPU                                                                                                                               
   */                                                                                                                                                                               
                                                                                                                                                                                    
  irqstate_t flags = enter_critical_section();                                                                                                                                      
#else                                                                                                                                                                               
  /* Disable pre-emption so that nothing changes while the loop executes */                                                                                                         
                                                                                                                                                                                    
  sched_lock();                                                                                                                                                                     
#endif 

...

#ifdef CONFIG_SMP                                                                                                                                                                   
  leave_critical_section(flags);                                                                                                                                                    
#else                                                                                                                                                                               
  sched_unlock();                                                                                                                                                                   
#endif   
```

Reply:
> I think replacing sched_lock() and sched_unlock() with a critical section should be a future task.

Discussed a little in issue #1138, Related to issue #1137